doc/citing.md

```diff
@@ -44,39 +44,47 @@ An asterisk (`*`) indicates co-equal lead authorship (listed alphabetically).
 
 > Moon, T. A., M. Fisher, T. Stafford, and A. Thurber (2023). QGreenland ({{ release }})
 > [dataset], [National Snow and Ice Data Center](https://qgreenland.org/).
+> doi: [10.5281/zenodo.12823307](https://doi.org/10.5281/zenodo.12823307)
```

Comment:
This is the bumpversion-driven citation section. The `{{ release }}` bit above is a myst substitution based on the software's current version number.

I hard-coded the DOI to v3, our latest release, but a future release will mean this is wrong.

We could use the "cite all versions" doi here, but it does mean that someone that copies and uses this citation, intending to get the v3 citation, might end up with a DOI that points to e.g., v4 in a year or two from now. Maybe that's not a huge concern, since the version number would still be in the citation?

@mfisher87 , any thoughts?